src/jit/lower.cpp

```diff
@@ -4384,34 +4392,6 @@ GenTree* Lowering::TryCreateAddrMode(LIR::Use&& use, bool isIndir)
     ssize_t  offset = 0;
     bool     rev    = false;
 
-    // TODO-1stClassStructs: This logic is here to preserve prior behavior. Note that previously
-    // block ops were not considered for addressing modes, but an add under it may have been.
-    // This should be replaced with logic that more carefully determines when an addressing mode
-    // would be beneficial for a block op.
-    if (isIndir)
-    {
-        GenTree* indir = use.User();
-        if (indir->TypeGet() == TYP_STRUCT)
-        {
-            isIndir = false;
-        }
-        else if (varTypeIsStruct(indir))
-        {
-            // We can have an indirection on the rhs of a block copy (it is the source
-            // object). This is not a "regular" indirection.
-            // (Note that the user check could be costly.)
-            LIR::Use indirUse;
-            if (BlockRange().TryGetUse(indir, &indirUse) && indirUse.User()->OperIsIndir())
```

Comment:
The removal of this code is responsible for the sole diff in this PR:
```diff
--- D:\Projects\jit\diffs\dasmset_16\base\System.Private.CoreLib.dasm	2019-10-04 20:51:01.000000000 +-0300
+++ D:\Projects\jit\diffs\dasmset_16\diff\System.Private.CoreLib.dasm	2019-10-04 20:51:01.000000000 +-0300
@@ -439051,6 +439051,5 @@
 G_M6933_IG02:
-       add      rdx, r8
-       movups   xmm0, xmmword ptr [rdx]
+       movups   xmm0, xmmword ptr [rdx+r8]
        movups   xmmword ptr [rcx], xmm0
        mov      rax, rcx
 ```
`ContainCheckHWIntrinsicAddr` calls `TryCreateAddrMode` with `isIndir = true` but then `TryCreateAddrMode` decides that it knows better...